cppForSwig/Makefile

```diff
@@ -17,7 +17,14 @@ endif
 platform=$(shell uname)
 
 ifeq ($(shell uname), Darwin)
+# XCode 8 breaks when compiling for 10.7, XCode 7 and below does not
+XCODE_VERSION:=$(shell xcodebuild -version | awk '{ print $$2 }' | head -n1 | cut -d "." -f 1)
+ifeq ($(shell echo "${XCODE_VERSION} >= 8" | bc), 1)
+$(info XCode 8 Detected. Due to an XCode bug, only 10.8+ targets are supported.)
```

Comment:
Okay. This works! I made a couple of changes that I'd like to recommend a couple of changes. I've pasted the code below.

```
# XCode 8 breaks when compiling for 10.7, XCode 7 and below does not
XCODE_VERSION_FULL:=$(shell xcodebuild -version | awk '{ print $$2 }')
XCODE_VERSION:=$(shell xcodebuild -version | awk '{ print $$2 }' | head -n1 | cut -d "." -f 1)
$(info XCode ${XCODE_VERSION_FULL} detected.)
ifeq ($(shell echo "${XCODE_VERSION} >= 8" | bc), 1)
$(info Due to XCode 8+ changes, the deployment target is 10.8. Use XCode 7 if a target of 10.7 is needed.)
MACOSX_DEPLOYMENT_TARGET=10.8
else
$(info A version of XCode below XCode 8 has been detected. The deployment target is 10.7.)
MACOSX_DEPLOYMENT_TARGET=10.7
endif
```